Bug: JVM options are not correctly set by jnius_config.set_options()

Create a new test file named `test_jvm_options.py` to the `tests` folder with the content below:
```
from __future__ import print_function
from __future__ import division
from __future__ import absolute_import
import pytest
import subprocess
import textwrap
import json


class TestJVMOptions:
    @pytest.mark.parametrize(
        "options",
        [
            [],
            ['-Dtest.var{}=value'.format(i) for i in range(2)],
            ['-Dtest.var{}=value'.format(i) for i in range(5)],
            ['-Dtest.var{}=value'.format(i) for i in range(10)],
            ['-Dtest.var{}=value'.format(i) for i in range(40)],
        ],
    )
    def test_jvm_options(self, options):
        process = subprocess.Popen(['python', '-c', textwrap.dedent(
            '''\
            import jnius_config
            import json
            import sys
            jnius_config.set_options(*json.load(sys.stdin))
            import jnius
            ManagementFactory = jnius.autoclass("java.lang.management.ManagementFactory")
            json.dump(list(ManagementFactory.getRuntimeMXBean().getInputArguments()), sys.stdout)''')],
            bufsize=-1, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
        stdoutdata, _ = process.communicate(json.dumps(options).encode())
        assert process.wait() == 0
        actual_options = json.loads(stdoutdata.decode())

        assert list(sorted(options)) == list(sorted(actual_options))
```
Run it with `pytest -vv test_jvm_options.py` command, and the output is:
```
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.8.3, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /home/user/.virtualenvs/jniusdbg/bin/python
cachedir: .pytest_cache
rootdir: /home/user/pyjnius
collected 5 items                                                                                                                                                                                                                            

test_jvm_options.py::TestJVMOptions::test_jvm_options[options0] PASSED                                                                                                                                                                 [ 20%]
test_jvm_options.py::TestJVMOptions::test_jvm_options[options1] PASSED                                                                                                                                                                 [ 40%]
test_jvm_options.py::TestJVMOptions::test_jvm_options[options2] FAILED                                                                                                                                                                 [ 60%]
test_jvm_options.py::TestJVMOptions::test_jvm_options[options3] FAILED                                                                                                                                                                 [ 80%]
test_jvm_options.py::TestJVMOptions::test_jvm_options[options4] FAILED                                                                                                                                                                 [100%]

================================================================================================================== FAILURES ==================================================================================================================
_________________________________________________________________________________________________ TestJVMOptions.test_jvm_options[options2] __________________________________________________________________________________________________

self = <test_jvm_options.TestJVMOptions object at 0x7efcb02eb460>, options = ['-Dtest.var0=value', '-Dtest.var1=value', '-Dtest.var2=value', '-Dtest.var3=value', '-Dtest.var4=value']

    @pytest.mark.parametrize(
        "options",
        [
            [],
            ['-Dtest.var{}=value'.format(i) for i in range(2)],
            ['-Dtest.var{}=value'.format(i) for i in range(5)],
            ['-Dtest.var{}=value'.format(i) for i in range(10)],
            ['-Dtest.var{}=value'.format(i) for i in range(40)],
        ],
    )
    def test_jvm_options(self, options):
        process = subprocess.Popen(['python', '-c', textwrap.dedent(
            '''\
            import jnius_config
            import json
            import sys
            jnius_config.set_options(*json.load(sys.stdin))
            import jnius
            ManagementFactory = jnius.autoclass("java.lang.management.ManagementFactory")
            json.dump(list(ManagementFactory.getRuntimeMXBean().getInputArguments()), sys.stdout)''')],
            bufsize=-1, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
        stdoutdata, _ = process.communicate(json.dumps(options).encode())
        assert process.wait() == 0
        actual_options = json.loads(stdoutdata.decode())
    
>       assert list(sorted(options)) == list(sorted(actual_options))
E       AssertionError: assert ['-Dtest.var0=value',\n '-Dtest.var1=value',\n '-Dtest.var2=value',\n '-Dtest.var3=value',\n '-Dtest.var4=value'] == ['-Dtest.var3=value',\n '-Dtest.var3=value',\n '-Dtest.var4=value',\n '-Dtest.var4=value',\n '-Dtest.var4=value']
E         At index 0 diff: '-Dtest.var0=value' != '-Dtest.var3=value'
E         Full diff:
E           [
E         -  '-Dtest.var3=value',
E         ?             ^
E         +  '-Dtest.var0=value',
E         ?             ^
E         +  '-Dtest.var1=value',
E         +  '-Dtest.var2=value',
E            '-Dtest.var3=value',
E            '-Dtest.var4=value',
E         -  '-Dtest.var4=value',
E         -  '-Dtest.var4=value',
E           ]

test_jvm_options.py:36: AssertionError
_________________________________________________________________________________________________ TestJVMOptions.test_jvm_options[options3] __________________________________________________________________________________________________
......
```

My best guess is the `s.encode(...)` inside the `str_for_c()` method reuses the allocated memory. But I don't have any proof for this yet.

--- a/jnius/jnius_jvm_desktop.pxi
+++ b/jnius/jnius_jvm_desktop.pxi
@@ -35,10 +35,9 @@ cdef void create_jnienv() except *:
     cp = jnius_config.expand_classpath()
     optarr.append("-Djava.class.path={0}".format(cp))
 
+    optarr = [str_for_c(x) for x in optarr]
     options = <JavaVMOption*>malloc(sizeof(JavaVMOption) * len(optarr))
     for i, opt in enumerate(optarr):
-        if PY_MAJOR_VERSION >= 3:
-           opt = opt.encode('utf-8')
         options[i].optionString = <bytes>(opt)
         options[i].extraInfo = NULL
 

--- a/jnius/jnius_jvm_dlopen.pxi
+++ b/jnius/jnius_jvm_dlopen.pxi
@@ -108,10 +108,10 @@ cdef void create_jnienv() except *:
     optarr = jnius_config.options
     optarr.append("-Djava.class.path=" + jnius_config.expand_classpath())
 
+    optarr = [str_for_c(x) for x in optarr]
     options = <JavaVMOption*>malloc(sizeof(JavaVMOption) * len(optarr))
     for i, opt in enumerate(optarr):
-        optbytes = str_for_c(opt)
-        options[i].optionString = <bytes>(optbytes)
+        options[i].optionString = <bytes>(opt)
         options[i].extraInfo = NULL
 
     args.version = JNI_VERSION_1_6

--- a/tests/test_jvm_options.py
+++ b/tests/test_jvm_options.py
@@ -1,0 +1,31 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+import json
+import pytest
+import subprocess
+import sys
+import textwrap
+
+
+class TestJVMOptions:
+    @pytest.mark.skipif(
+        sys.platform == 'android',
+        reason='JNIus on Android does not take JVM options'
+    )
+    def test_jvm_options(self):
+        options = ['-Dtest.var{}=value'.format(i) for i in range(40)]
+        process = subprocess.Popen(['python', '-c', textwrap.dedent(
+            '''\
+            import jnius_config
+            import json
+            import sys
+            jnius_config.set_options(*json.load(sys.stdin))
+            import jnius
+            ManagementFactory = jnius.autoclass("java.lang.management.ManagementFactory")
+            json.dump(list(ManagementFactory.getRuntimeMXBean().getInputArguments()), sys.stdout)''')],
+            bufsize=-1, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        stdoutdata, _ = process.communicate(json.dumps(options).encode())
+        assert process.wait() == 0
+        actual_options = json.loads(stdoutdata.decode())
+        assert list(sorted(options)) == list(sorted(actual_options))

